### PR TITLE
grid: real ship display

### PR DESCRIPTION
**Pull request by arthyn**
This pulls over contact store from current Landscape and adds a new component for rendering sigils/avatars.

Two issues currently, there's a bad module ref in urbit/sigil-js that needs to be resolved looks like a mis-paste. Second there's some eslint issues with some of the contact store code. Other than that seems to work on mock data and testnet.

### CODE REVIEW

File: pkg/grid/src/nav/search/Providers.tsx [hunk 1]
```diff
@@ -39,7 +41,7 @@ export const Providers = ({ match }: ProvidersProps) => {
 
               return right - left;
             })
-            .map((el) => ({ shipName: el.original }))
+            .map((el) => ({ shipName: el.original, ...contacts[el.original] }))
         : [],
     [allies, search]
```
Comment (liam-fitzgerald) [inline pkg/grid/src/nav/search/Providers.tsx:44]: Missing contacts as a dep

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
LGTM! I'll make these changes bc you're out this week

**Review by liam-fitzgerald — APPROVED** [review]

**Comment by liam-fitzgerald** [comment]
Blocked on design feedback and also of new release of `@tlon/sigil-js` cc: @urcades @g-a-v-i-n

**Comment by urcades** [comment]
For an issue like this it might hasten review time if we could get a screenshot of the (presumably) single place sigils/avatars are displayed — right now I'm running into build issues and can't seem to get `esbuild` related errors to go away after following the error debugging prompts:

```
failed to load config from /Users/ed/Urbit Projects/urbit/pkg/grid/vite.config.ts
error when starting dev server:
Error: The service was stopped
    at /Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:1215:25
    at /Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:609:9
    at Socket.afterClose (/Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:587:7)
    at Socket.emit (node:events:406:35)
    at Socket.emit (node:domain:475:12)
    at endReadableNT (node:internal/streams/readable:1343:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

**Comment by urcades** [comment]
making some progress here after trying to reset package-lock and node_modules:

```
 > src/components/Avatar.tsx:3:37: error: [plugin: vite:dep-scan] Failed to resolve entry for package "@tlon/sigil-js". The package may have incorrect main/module/exports specified in its package.json: Failed to resolve entry for package "@tlon/sigil-js". The package may have incorrect main/module/exports specified in its package.json.
 ```

and this, which follows:

```
error when starting dev server:
Error: Build failed with 1 error:
src/components/Avatar.tsx:3:37: error: [plugin: vite:dep-scan] Failed to resolve entry for package "@tlon/sigil-js". The package may have incorrect main/module/exports specified in its package.json: Failed to resolve entry for package "@tlon/sigil-js". The package may have incorrect main/module/exports specified in its package.json.
    at failureErrorWithLog (/Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:1451:15)
    at /Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:1131:28
    at runOnEndCallbacks (/Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:921:63)
    at buildResponseToResult (/Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:1129:7)
    at /Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:1238:14
    at /Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:609:9
    at handleIncomingPacket (/Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:706:9)
    at Socket.readFromStdout (/Users/ed/Urbit Projects/urbit/pkg/grid/node_modules/esbuild/lib/main.js:576:7)
    at Socket.emit (node:events:394:28)
    at Socket.emit (node:domain:475:12)
```

**Comment by g-a-v-i-n** [comment]
I published @tlon/sigil-js 1.4.4 which
    * Corrects typo in package.json
    * Updates react dep to ^17

Hope that helps

**Comment by g-a-v-i-n** [comment]
And yes, screen shots would be greatly appreciated

**Review by g-a-v-i-n — APPROVED** [review]

**Pull request merged**